### ISSUE TITLE
how to delete calendar

### ISSUE THREAD

**Issue by maxandersen** [python-caldav/caldav#7]
I can successfully create a calendar via `make_calendar` but I have not found a way to delete a calendar.

calling `cal.delete()` result in an error.

I do this:
```
client = caldav.DAVClient(url)
principal = client.principal()
calendars = principal.calendars()

for c in calendars:
    if c.name == 'orgtodo':
        c.delete()
```

which gives me:
```
Login to https://xhz:secret@p01-caldav.icloud.com:443/
Traceback (most recent call last):
  File "test.py", line 38, in <module>
    c.delete()
  File "/usr/local/lib/python2.7/site-packages/caldav/objects.py", line 238, in delete
    raise error.DeleteError(errmsg(r))
caldav.lib.error.DeleteError: 400 Bad Request

<html><head><title>Bad Request</title></head><body><h1>Bad Request</h1><p>Client sent illegal depth header value for DELETE: 1</p></body></html>
```

Is this icloud specific or is delete broken or not the right way to do it  ?

**Comment by tobixen** [comment]
Sorry the slow response, seems like I've forgotten to set up email notifications on new issues here.

This is probably related to https://github.com/python-caldav/caldav/issues/3

**Comment by kyloe** [comment]
Have investigated here for iCloud (I guess this will be representative) 

Looks like caldav 0.5 sends a 'Depth' header on DELETE requests, which iCloud objects to.

instead of 
  
  calendar.delete()

try this code fragment as a work around

        del calendar.client.headers['Depth']
        calendar.client.request(calendar.url,'DELETE')

Works for iCloud - will make a pull request with some other stuff also in the near future

**Comment by tobixen** [comment]
I'm looking into this now.  The dirty-naïve solution, just add that "del client.headers['Depth']" in the request caused test failures, besides it feels completely wrong.

Looking a bit into the code, I can't understand how 'Depth' got into the client headers at all.  It does not belong there.  Hm.

**Comment by tobixen** [comment]
I threw in a temporary assert that there should be no Depth in client.headers, and the test code passes.  No depth header should be passed in the delete request.  I don't understand this.

**Comment by tobixen** [comment]
Seems related to #3

**Comment by tobixen** [comment]
I assume this bug was fixed with 72a089cc0d33207a34acd6fbfd4d72ed0c556e49.  Closing.
